web/frontend/src/pages/form/components/ActionButtons/CombineButton.tsx

```diff
@@ -3,17 +3,22 @@ import { AuthContext } from 'index';
 import { useContext } from 'react';
 import { useTranslation } from 'react-i18next';
 import { OngoingAction, Status } from 'types/form';
-import { UserRole } from 'types/userRole';
 import ActionButton from './ActionButton';
 
+function hasAuthorization(authCtx, subject: string, action: string): boolean {
```

Comment:
This function is duplicated a lot. This would be a good idea to extract it somewhere and write it only once. In fact, it could be part of the `authCtx`. Start by updating the `AuthState` interface as follow: 

```ts
export interface AuthState {
  isLogged: boolean;
  firstname: string;
  lastname: string;
  role: string;
  authorization: Map<String, Array<String>>;
  isAllowed: (subject: string, action: string) => boolean;
}
```

then implement the `isAllowed` function:

```ts
const defaultAuth = {
  isLogged: false,
  firstname: '',
  lastname: '',
  role: '',
  authorization: arr,
  isAllowed: function (): boolean {
    return false;
  },
};

...

        setAuth({
          isLogged: result.islogged,
          firstname: result.firstname,
          lastname: result.lastname,
          role: result.role,
          authorization: result.islogged ? new Map(Object.entries(result.authorization)) : arr,
          isAllowed: function (subject: string, action: string): boolean {
            return (
              this.authorization.has(subject) &&
              this.authorization.get(subject).indexOf(action) !== -1
            );
          },
        });
```